### ISSUE TITLE
useage of args.first_order

### ISSUE THREAD

**Issue by battlezzq** [kdwonn/DivE#2]
from your codebase release on the page https://cvlab.postech.ac.kr/research/DivE/
there the if i == len(self.layers) - 1 and self.first_order:
                    x = x.detach()
i wander when the first_order will be set into True? and why ?   Thanks a lot !!!
        for i, (cross_attn, pn1, cross_ff, pn2) in enumerate(self.layers):
            if i == len(self.layers) - 1 and self.first_order:
                x = x.detach()
            x = cross_attn(x, context = local_feat, mask = mask, k_pos = pos, q_pos = None) + x
            x = pn1(x)
            x = cross_ff(x) + x
            x = pn2(x)

**Comment by kdwonn** [comment]
Sorry for the late reply. 

first_order option enables the model to be trained with implicit differentiation, instead of backproped through the multiple iterations. You can find detailed explanation for the implicit differentiation of slot-attention in here: [https://arxiv.org/abs/2207.00787](https://arxiv.org/abs/2207.00787 ).

It is known to be helpful for efficiency and accuracy, but unfortunately, not in our case. I removed it in the repository since I hadn't used this option during the experiments.
